core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -188,6 +217,120 @@ public SimpleQueryBus(MessageMonitor<? super QueryMessage<?, ?>> messageMonitor,
                        .filter(Objects::nonNull);
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <Q, I, U> Registration subscriptionQuery(SubscriptionQueryMessage<Q, I, U> query,
+                                                    UpdateHandler<I, U> updateHandler) {
+        MessageMonitor.MonitorCallback monitorCallback = messageMonitor.onMessageIngested(query);
+        SubscriptionQueryMessage<Q, I, U> interceptedQuery = intercept(query);
+        List<QuerySubscription> subs = subscriptions
+                .computeIfAbsent(interceptedQuery.getQueryName(), k -> new CopyOnWriteArrayList<>())
+                .stream()
+                .filter(subscription -> interceptedQuery.getResponseType().matches(subscription.getResponseType()))
+                .collect(Collectors.toList());
+        Registration registration = () -> true;
+        try {
+            if (subs.isEmpty()) {
+                throw new NoHandlerForQueryException(
+                        format("No handler found for %s with response type %s and update type %s",
+                               interceptedQuery.getQueryName(),
+                               interceptedQuery.getResponseType(),
+                               interceptedQuery.getUpdateResponseType()));
+            }
+            Iterator<QuerySubscription> subsIterator = subs.iterator();
+            boolean invocationSuccess = false;
+            while (!invocationSuccess && subsIterator.hasNext()) {
+                try {
+                    QuerySubscription subscription = subsIterator.next();
+                    DefaultUnitOfWork<QueryMessage<Q, I>> uow = DefaultUnitOfWork.startAndGet(interceptedQuery);
+                    if (subscription instanceof SubscribableQuerySubscription) {
```

Comment:
It's definitely a matter of taste: I would leave it like this, since first `if` expresses the difference between two cases we wanna cover here (subscribable and regular query subscription). If we merge them we would have to introduce new `if` to the `else` branch making sure that we are not dealing with subscribable query subscription.

Reply:
Sounds fair!